weather/src/js/components/WeatherApp.jsx

```diff
@@ -0,0 +1,94 @@
+import React from 'react';
+import { Table,td,tr,th,thead,tbody } from 'react-bootstrap';
+import { getLocationCoords, getWeatherData } from '../api/weather-api';
+
+export default class WeatherApp extends React.Component {
+
+  constructor() {
+    super(); // A call to the class we are extending: React.Component
+
+    this.state = {
+      data : {
+         forecastList : [],
+         city: 'Loading...',
+         country: 'Loading...'
+      }
+    };
+
+    this.componentDidMount = this.componentDidMount.bind(this);
+  }
+  
+  componentDidMount() {
+        this.fetchWeather(this.state.currentUnit);
+  }
+ 
+  fetchWeather(units) {
+      getLocationCoords().then(
+        (coords) => {
+          // Got location data, get weather now
+          getWeatherData(units, coords).then(
+            (data) => {
+                this.setState({data:data});
+            }, (error) => {
+              // Could not get weather data.
+              console.error(error);
+            }
+          );
+        }, (error) => {
+          // Could not get location data.
+          console.error(error);
+        }
+      );
+
+  }
+  render() {
+      let lst = this.state.data.forecastList;
+      return (
```

Comment:
I would split the render body into 3 calls. 

1. `renderHeader`
2. `renderBody`